### PR TITLE
fix for Bug 60509

### PR DESCRIPTION
**Pull request by Alain-Bearez**
https://bz.apache.org/bugzilla/show_bug.cgi?id=60509

All the code was provided by Guillaume de GENTILE in https://bz.apache.org/bugzilla/show_bug.cgi?id=54470, only waiting for the XDDF package to be ready, since 2015.

### CODE REVIEW

**Comment by asfgit** [comment]
Can one of the admins verify this patch?

File: src/ooxml/java/org/apache/poi/xddf/usermodel/chart/XDDFDataSourcesFactory.java [hunk 1]
```diff
@@ -41,46 +41,97 @@ private XDDFDataSourcesFactory() {
...
-                return categoryDS.getStrRef().getF();
-            }
-
-            @Override
-            public int getColIndex() {
-                return 0;
-            }
-        };
+        if (categoryDS.getStrRef() == null) {
+            return new XDDFCategoryDataSource() {
```
Comment (pjfanning) [inline src/ooxml/java/org/apache/poi/xddf/usermodel/chart/XDDFDataSourcesFactory.java:45]: most of the code in the 2 anonymous classes is the same - would it be possible to create an inner class that has the shared code and to have the 2 anonymous classes extend from it?

File: src/ooxml/java/org/apache/poi/xddf/usermodel/chart/XDDFDataSourcesFactory.java [hunk 2]
```diff
@@ -41,46 +41,97 @@ private XDDFDataSourcesFactory() {
...
+                }
+
+                @Override
+                public String getPointAt(int index) {
+                    return category.getPtArray(index).getV();
+                }
+
+                @Override
+                public String getDataRangeReference() {
+                    return categoryDS.getStrRef().getF();
```
Comment (Alain-Bearez) [inline src/ooxml/java/org/apache/poi/xddf/usermodel/chart/XDDFDataSourcesFactory.java]: Note to self: should have been `getNumRef` here too! Remember to change that during the refactoring.

File: src/ooxml/java/org/apache/poi/xddf/usermodel/chart/XDDFDataSourcesFactory.java (hunk 1 above)
Comment (Alain-Bearez) [reply src/ooxml/java/org/apache/poi/xddf/usermodel/chart/XDDFDataSourcesFactory.java]: For sure!

Comment (Alain-Bearez) [reply src/ooxml/java/org/apache/poi/xddf/usermodel/chart/XDDFDataSourcesFactory.java:45]: Like often with the OOXML code, it looks the same but has a different underlying class:
 `category` is either `CTNumData` or `CTStrData`... For the rest, I extracted the default implementation to the interface level.

**Review by pjfanning — COMMENTED** [review]
LGTM

**Comment by Alain-Bearez** [comment]
fixed by r1842548

**Pull request closed without merging**